docs/models_and_schedulers.rst

```diff
@@ -252,3 +252,146 @@ the following happens:
   requests.
 
 .. image:: images/sequence_example2.png
+
+.. _section-ensemble-models:
+
+Ensemble Models
+---------------
+
+An ensemble model represents a *pipepline* of one or more models and the
+connection of input and output tensors between those models. Ensemble models are
+intended to be used to encapsulate a procedure that involves
+multiple models, such as "data preprocessing -> inference -> data postprocessing".
+Using ensemble models for this purpose can avoid the overhead of transferring
+intermediate tensors and minimize the number of requests that must be sent to
+the inference server.
+
+The :ref:`ensemble scheduler <section-ensemble-scheduler>` must be used for
+ensemble models, regardless of the scheduler used by the models within the
+ensemble. With respect to the ensemble scheduler, an *ensemble* model is not
+an actual model. Instead, it specifies the dataflow between models within the
+ensemble as :cpp:var:`Step
+<nvidia::inferenceserver::ModelEnsembling::Step>`. The
+scheduler collects the output tensors in each step, provides them as input
+tensors for other steps according to the specification. In spite of that, the
+ensemble model is still viewed as a single model from an external view.
+
+Note that the ensemble models will inherit the characteristics of the models
+involved, so the meta-data in the request header must comply with the models
+within the ensemble. For instance, if one of the models is stateful model, then
+the inference request for the ensemble model should contain the information
+mentioned in the previous :ref:`section <section-stateful-models>`, which will
+be provided to the stateful model by the scheduler. This is a limitation of
+ensemble where the stateful model in the ensemble must share the same flag.
+
+As a running example, consider an ensemble model for image classification and
+segmentation that has the following model configuration::
+
+  name: "ensemble_model"
+  platform: "ensemble"
+  max_batch_size: 1
+  input [
+    {
+      name: "IMAGE"
+      data_type: TYPE_STRING
+      dims: [ 1 ]
+    }
+  ]
+  output [
+    {
+      name: "CLASSIFICATION"
+      data_type: TYPE_FP32
+      dims: [ 1000 ]
+      label_filename: "classification_labels.txt"
```

Comment:
We shouldn't need the label file here for the ensemble (pending your other fix)

Reply:
Removed. And addressed the other comment